manual/debugging/find-s3-upload-folder.md

```diff
@@ -0,0 +1,51 @@
+# Find S3 upload folder
+
+When you upload files to TDR, they are saved to the dirty bucket
+(`tdr-upload-files-dirty-<environment>`). Within the bucket, the S3 object key
+structure is like this: `<cognito-user-id>/<consignment-id>/<file-id>`, e.g.
+`eu-west-2:56f64881-67ca-4657-87dc-3065a6ce3b20/01573ee9-05f6-4a1c-9bd4-b26020bebe1a/764996b2-57cc-431e-b6f2-65ae4ecda3fa`.
+
+The reason for the top-level folder based on the Cognito user ID is that it lets
+us set user-specific access permissions. See [ADR 7] for more details.
+
+The folder structure does make it tricky to find a specific file in S3, because
+you have to know all three IDs. You can find the consignment ID from the URL in
+TDR. You can look up the file IDs in the database or monitor the API requests
+made by the frontend JavaScript when it saves the file metadata.
+
+The Cognito user ID is much harder to work out, because it is not stored in TDR.
+It is generated by the frontend JavaScript to use when authenticating the file
+upload requests.
+
+Unfortunately, we don't know of a simple way to map between all Keycloak and
+Cognito IDs, but you can look up the Cognito ID for your own test user.
+
+## Option 1: Browser local storage
+
```

Comment:
I've tried this in Chrome and Firefox on Ubuntu and I'm not seeing anything in local storage. This might be just something dodgy with my machine but is this still working for you?

Reply:
That's weird. This is what I've got in Firefox:
<img width="1170" alt="image" src="https://user-images.githubusercontent.com/754712/122014738-5d1fb380-cdb7-11eb-8c3e-9d466ce03037.png">

But I just tried it in a private window, and it's not there. 🤔 

Is there anything that's changed in the past year or so that would have switched from using local storage to something else?

Reply:
The only thing I can think of is that they've finally stopped using local storage for things and we've included this in one of the dependabot upgrades but I can't find anything on Google and you think they'd mention it.

Reply:
Either that or it only stores the value in local storage under certain conditions.

I've removed that option, anyway. The AWS CLI option is pretty straightforward if you already some way of getting auth tokens, like with Postman.